[WIP]ChatspaceのREADMEにテーブル設計を追加

northernblacksmith opened the pull request:
# WHAT
READMEにテーブル設計を追加

# WHY
他の開発者がテーブル設計を理解できるよう明示するため。

a1exkita committed:
edit readme add table

taketnk commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|------|----|-------|
+|member_id|integer|null :false, foreign_key: true|
+|message|string|null :false, foreign_key: true|
+|created_at|timestamp|null :false, foreign_key: true|
```
デフォルトであるので書かなくて結構です。

taketnk commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|------|----|-------|
+|name|string|null :false, foreign_key: true|
+|mail|string|null :false, foreign_key: true|
+|group_id|integer|null :false, foreign_key: true|
```
不要ですね。

taketnk commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+
+### Association
+-has_many :messages
+-has_many :groups, through :members
```
アソシエーションが一つたりてないです。

taketnk commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|column|type|options|
+|------|----|-------|
+|group_name|string|null :false, foreign_key: true|
+|members_id|integer|null :false, foreign_key: true|
```
不要ですね。

taketnk commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|------|----|-------|
+|group_name|string|null :false, foreign_key: true|
+|members_id|integer|null :false, foreign_key: true|
+|messages|string|null :false, foreign_key: true|
```
いりませんね。

taketnk commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+|messages|string|null :false, foreign_key: true|
+
+### Association
+-has_many :users, through :members
```
アソシエーションが一つたりてないです

taketnk commented on README.md:
```diff
@@ -22,3 +22,52 @@ Things you may want to cover:
+## messagesテーブル
+|column|type|options|
+|------|----|-------|
+|member_id|integer|null :false, foreign_key: true|
```
不要ですね。

a1exkita committed:
fixed parts of pointed in README

・member_idをカラムにしないように訂正。
・多対多のアソシエーションの書き方を修正。
・foreign_keyが不要だった部分を削除。

taketnk commented on README.md:
```diff
@@ -27,24 +27,24 @@ Things you may want to cover:
-|members_id|integer|null :false, foreign_key: true|
-|messages|string|null :false, foreign_key: true|
+|group_name|string|null :false|
+|messages|string|null :false|
```
このカラムはいりません。

taketnk commented on README.md:
```diff
@@ -53,20 +53,20 @@ Things you may want to cover:
-|created_at|timestamp|null :false, foreign_key: true|
+|user_id|integer|null :false, foreign_key: true|
+|group_id|integer|null :false, foreign_key: true|
+|message|string|null :false|
```
写真も投稿できるので、imageカラムも必要です。

taketnk commented on README.md:
```diff
@@ -53,20 +53,20 @@ Things you may want to cover:
--belongs_to :group, through :member
+- belongs_to :user, through :member
+- belongs_to :group, through :member
+- belongs_to :member
```
messageでは、throughオプションは不要です。

messageのbelongs_to :member不要です。

taketnk commented on README.md:
```diff
@@ -27,24 +27,24 @@ Things you may want to cover:
-|name|string|null :false, foreign_key: true|
-|mail|string|null :false, foreign_key: true|
-|group_id|integer|null :false, foreign_key: true|
+|name|string|null :false|
```
nameが一意になるような制約もかけておきましょう。
加えて、indexもはっておいたほうたが良いです。

a1exkita committed:
fixed parts of pointed ver.2

# 修正箇所
１userテーブルのnameカラムに、一意性制約&index
２groupテーブルからgroup_nameカラムを削除
３messageテーブルにimageカラムを追加
４messageテーブルのアソシエーションから、memberテーブルを介する表記を削除

taketnk commented on README.md:
```diff
@@ -22,3 +22,51 @@ Things you may want to cover:
+|------|----|-------|
+|user_id|integer|null :false, foreign_key: true|
+|group_id|integer|null :false, foreign_key: true|
+|message|string|null :false|
```
これだと、imageだけの送信ができなくなってしまうので、null: falseは外しておきましょう。

taketnk commented on README.md:
```diff
@@ -22,3 +22,51 @@ Things you may want to cover:
+## groupsテーブル
+|column|type|options|
+|------|----|-------|
+|messages|string|null :false|
```
groupsテーブルにほぞんするのは、groupなので、messageというカラム名よりも、nameとかのほうがふさわしい気がします。
